[@i18next-selector/vite-plugin] Cannot destructure property 'jsdocs' of 'x.left' as it is undefined

Hi, I tried to use new `enableSelector` api and it doesn't work with my repo. 

First is `codemod`:

```
❯ npx i18next-selector-codemod
node:internal/modules/cjs/loader:1413
  throw err;
  ^

Error: Cannot find module '####\node_modules\npm\bin\npm-prefix.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1410:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1051:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1056:22)
    at Module._load (node:internal/modules/cjs/loader:1219:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:238:24)
    at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:154:5)
    at node:internal/main/run_main_module:33:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v24.6.0
node:internal/modules/cjs/loader:1413
  throw err;
  ^

Error: Cannot find module '####\node_modules\npm\bin\npx-cli.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1410:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1051:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1056:22)
    at Module._load (node:internal/modules/cjs/loader:1219:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:238:24)
    at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:154:5)
    at node:internal/main/run_main_module:33:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v24.6.0
```

I found a way to run it, but it still doesn't work with my setup:
```
npx jscodeshift --transform="node_modules\i18next-selector-codemod\dist\esm\transform.js" --ignore-pattern="**/node_modules/**" --ignore-pattern="*.d.ts" --keySeparator='::' --nsSeparator=':::' --parser=tsx ./src
Processing 305 files...
Spawning 23 workers...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 14 files to free worker...
Sending 11 files to free worker...
All done.
Results:
0 errors
304 unmodified
0 skipped
1 ok
Time elapsed: 2.007seconds
```

And... it doesn't change anything (1 ok file has no changes, only encoding). I'm ok to change translations manually, and finally tried to use vite plugin:

astro.config.ts:
```ts
...
  vite: {
    plugins: [
      tailwindcss(),
      i18nextVitePlugin({
        sourceDir: './src/lib/i18n/locales',
        silent: false,
      }),
    ],
...
```

error:

```
[@i18next-selector/vite-plugin]:
 source file detected: src\lib\i18n\locales\en.ts 

Cannot destructure property 'jsdocs' of 'x.left' as it is undefined.
  Location:
    ####\node_modules\@i18next-selector\vite-plugin\dist\esm\plugin.js:91:17
  Stack trace:
    at file:///####/node_modules/@i18next-selector/vite-plugin/dist/esm/plugin.js:91:17
    at file:///####/node_modules/@i18next-selector/vite-plugin/dist/esm/plugin.js:80:77
    at stringifyTypeScript (file:///####/node_modules/@i18next-selector/vite-plugin/dist/esm/plugin.js:100:5)
    at tsFileToDeclarationFile (file:///####/node_modules/@i18next-selector/vite-plugin/dist/esm/plugin.js:163:48)
    at Array.forEach (<anonymous>)
```

Even if I delete all  js docs comments in my translation files plugin is crashed with the same message.

With `enableSelector: true` ide works at least, but it's very unclear how to use all of this.


---
upd1:

I also use pluralSeparator/contextSeparator which can't be configured with codemod

```ts
    nsSeparator: ':::';
    keySeparator: '::';
    pluralSeparator: '##';
    contextSeparator: '##';
```


upd2:

translation files uses typescript with ~this structure:

```ts
const zod = {/* translations */} as const;
const component1 = {/* translations */} as const;

const translation = {
 ...zod,
 ...component1
};

export default translation;
```

--- a/.changeset/blue-lemons-hunt.md
+++ b/.changeset/blue-lemons-hunt.md
@@ -1,0 +1,6 @@
+---
+"@i18next-selector/vite-plugin": patch
+"i18next-selector-codemod": patch
+---
+
+fix(vite-plugin): adds support for custom, non-identifier syntax in pluralSeparator

--- a/packages/codemod/src/transform.ts
+++ b/packages/codemod/src/transform.ts
@@ -372,11 +372,9 @@ export function transform(
           selectorFn = j.arrowFunctionExpression(
             [j.identifier("$")],
             tokensToSelector(tokens.path, j)
-          )
-          console.log('HERE', tokens);
+          );
 
           (i18nAttr as { value: unknown }).value = j.jsxExpressionContainer(selectorFn)
-          // const ns = 
 
           if (tokens.ns && !attrs.some(a => a.type === "JSXAttribute" && a.name.name === "ns")) {
             attrs.push(j.jsxAttribute(j.jsxIdentifier("ns"), j.literal(tokens.ns)))

--- a/packages/vite-plugin/package.json
+++ b/packages/vite-plugin/package.json
@@ -44,6 +44,7 @@
     "test": "vitest"
   },
   "devDependencies": {
+    "@prettier/sync": "^0.6.1",
     "vite": "^6.3.5"
   },
   "dependencies": {

--- a/packages/vite-plugin/src/plugin.ts
+++ b/packages/vite-plugin/src/plugin.ts
@@ -175,7 +175,7 @@ export function transformTypeScript(
       case ts.isPropertyAssignment(x): {
         const children = x.getChildren()
         const jsdoc = children.find(ts.isJSDoc)?.getText() ?? null
-        const key = children.find(ts.isIdentifier)?.escapedText
+        const key = children[0].getText()
         const valueIndex = children.findIndex(ts.isColonToken) + 1
         const value = go(children[valueIndex], offset)
         return { jsdoc, key, value }

--- a/packages/vite-plugin/src/utils.ts
+++ b/packages/vite-plugin/src/utils.ts
@@ -67,7 +67,14 @@ export function rmPluralization(entry: [string, unknown], options: transform.Opt
   else {
     const [key] = entry
     const index = key.lastIndexOf(options.pluralSeparator ?? defaultOptions.pluralSeparator)
-    return just(key.slice(0, index))
+    let sliced = key.slice(0, index)
+    if (
+      sliced.startsWith('"') && !sliced.endsWith('"')
+      || sliced.startsWith("'") && !sliced.endsWith("'")
+    ) {
+      sliced = `${sliced}${sliced[0]}`
+    }
+    return just(sliced)
   }
 }
 

--- a/packages/vite-plugin/test/transform.test.ts
+++ b/packages/vite-plugin/test/transform.test.ts
@@ -1,6 +1,7 @@
 import * as vi from 'vitest'
 import * as path from 'node:path'
 import * as fs from 'node:fs'
+import prettier from '@prettier/sync'
 
 import {
   groupPluralKeys,
@@ -9,6 +10,8 @@ import {
   transformToTypeScript,
   writeFromSource,
 } from '@i18next-selector/vite-plugin'
+
+const format = (src: string) => prettier.format(src, { parser: 'typescript' })
 
 const defaultOptions = { contextSeparator: '_', pluralSeparator: '_' }
 
@@ -290,6 +293,43 @@ vi.describe('〖⛳️〗‹‹‹ ❲@i18next-selector/vite-plugin❳', () => {
   vi.it('〖⛳️〗› ❲transformToTypeScript❳', () => {
     vi.expect(transformToTypeScript(input, defaultOptions)).toMatchInlineSnapshot
       (`"export declare const resources: { "beverage": "beverage","beverage|beer": "beer","tea": "a cuppa tea and a lie down" | "two cups of tea" | "{{count}} cups of tea and a big sleep" | "many cups of tea","dessert|cake": "a nice cake","dessert|muffin": "a nice muffin" | "{{count}} nice muffins","coffee": { "drip": { "black": "a strong cup of black coffee" },"bar": { "shot": "a shot of espresso","espresso|americano": "a hot americano","espresso|latte": "a foamy latte" | "{{count}} foamy lattes","espresso|cappuccino": "a dry cappuccino" | "two dry cappuccinos" | "{{count}} dry cappuccinos" | "many dry cappuccinos" } },"sodas": { "coca_cola": { "coke": "a can of coke","coke|diet": "a can of diet coke" | "{{count}} cans of diet coke" },"faygo": { "purple": "purple faygo","orange": "one orange faygo" | "{{count}} orange faygo" } },"interpolation": { "val": "Interpolated {{val}}" },"array": ["element one",{ "elementTwo": "element two" },{ "elementThree": [{ "nestedElementThree": "element three" }] }] }"`)
+
+    vi.expect.soft(format(
+      transformToTypeScript({
+        time: 'время',
+        since: {
+          now: 'прямо сейчас',
+          ago: 'назад',
+          'seconds##one': '{{count}} секунду',
+          'seconds##few': '{{count}} секунды',
+          'seconds##many': '{{count}} секунд',
+          'minutes##one': '{{count}} минуту',
+          'minutes##few': '{{count}} минуты',
+          'minutes##many': '{{count}} минут',
+          'hours##one': '{{count}} час',
+          'hours##few': '{{count}} часа',
+          'hours##many': '{{count}} часов',
+          'days##one': '{{count}} день',
+          'days##few': '{{count}} дня',
+          'days##many': '{{count}} дней',
+        },
+      }, { pluralSeparator: '##' }
+      )
+    )).toMatchInlineSnapshot
+      (`
+      "export declare const resources: {
+        time: "время";
+        since: {
+          now: "прямо сейчас";
+          ago: "назад";
+          seconds: "{{count}} секунду" | "{{count}} секунды" | "{{count}} секунд";
+          minutes: "{{count}} минуту" | "{{count}} минуты" | "{{count}} минут";
+          hours: "{{count}} час" | "{{count}} часа" | "{{count}} часов";
+          days: "{{count}} день" | "{{count}} дня" | "{{count}} дней";
+        };
+      };
+      "
+    `)
   })
 
   vi.it('〖⛳️〗› ❲writeFromSource❳', () => {

